src/commands/apphosting-secrets-describe.ts

```diff
@@ -0,0 +1,29 @@
+import { Command } from "../command";
+import { Options } from "../options";
+import { needProjectId } from "../projectUtils";
+import { logger } from "../logger";
+import { requireAuth } from "../requireAuth";
+import { listSecretVersions } from "../gcp/secretManager";
+import * as secretManager from "../gcp/secretManager";
+import { requirePermissions } from "../requirePermissions";
+
+const Table = require("cli-table");
+
+export const command = new Command("apphosting:secrets:describe <secretName>")
+  .description("Get metadata for secret and its versions.")
+  .before(requireAuth)
+  .before(secretManager.ensureApi)
+  .before(requirePermissions, ["secretmanager.secrets.get"])
+  .action(async (secretName: string, options: Options) => {
+    const projectId = needProjectId(options);
+    const versions = await listSecretVersions(projectId, secretName);
+
+    const table = new Table({
+      head: ["Name", "Version", "Status", "Create Time"],
+      style: { head: ["yellow"] },
+    });
+    for (const version of versions) {
+      table.push([secretName, version.versionId, version.state, version.createTime]);
+    }
+    logger.info(table.toString());
```

Comment:
Applies for all 3 commands: These command should support the global `--json` flag, so that they can easily be used as part of scripts. To do this, you just need to return a JSON representation of the values displayed.

See https://github.com/firebase/firebase-tools/blob/master/src/commands/ext-dev-list.ts for an example of what this looks like